src/app/workflow/info-tab/info-tab.service.spec.ts

```diff
@@ -59,4 +59,14 @@ describe('ValueService', () => {
     expect(service.getTRSId(sampleWorkflow3, EntryType.Service)).toBe(`#service/${fullWorkflowPath}`);
     expect(service.getTRSId(null, EntryType.BioWorkflow)).toBe('');
   });
+
+  it(`getTRSPlainType should work for various descriptor types`, () => {
+    service = TestBed.inject(InfoTabService);
+    expect(service.getTRSPlainType('CWL')).toBe('PLAIN_CWL');
+    expect(service.getTRSPlainType('WDL')).toBe('PLAIN_WDL');
+    expect(service.getTRSPlainType('NFL')).toBe('PLAIN_NFL');
+    expect(service.getTRSPlainType('gxformat2')).toBe('PLAIN_GALAXY');
+    expect(service.getTRSPlainType('jupyter')).toBe('PLAIN_JUPYTER');
+    expect(service.getTRSPlainType('bogus')).toBe(null);
```

Comment:
If this was webservice code, we'd probably throw.  However, IMHO, in the UI, we probably want to continue on and do the best we can.  There's an interesting in-between solution, where if the UI detects an error condition, it can try to continue, but also log the problem somewhere (not necessarily the console), so we can check at the end of the test(s) to make them fail, or use the information for debugging purposes.

And, in that vein, rather than `null`, it probably should fall back to the the old way (capitalize and prepend `PLAIN_`) if using `descriptorTypeCompatService` fails...

Reply:
To check my assumptions, this error should only occur when we change the UI. The user can't really trigger the null condition on their own?

Reply:
> To check my assumptions, this error should only occur when we change the UI. The user can't really trigger the null condition on their own?

`getTRSPlainType` is used by `getTRSLink`, which is called by the tool and workflow info tab components to generate the TRS link. The case where `getTRSPlainType` would be error-ish is if we passed a Dockstore descriptor type that the underlying call to `descriptorTypeCompatService.stringToDescriptorType` couldn't convert. This would probably happen because we didn't add an entry for the descriptor type to `extendedDescriptorLanguages`, or didn't define it correctly.

Reply:
> Another idea regarding logging is that when we detected error-ish behavior on the client side, we could log it, and then periodically send the logs back to the webservice so that we'd know about it...

At my last job, we used to do this, albeit we collected said data with Google Analytics, not via direct calls to the backend, and would periodically do things like take a look at the "top ten" most common errors and try to pare them down.

Reply:
I added some code to generate a reasonable guess at the TRS plain type, in the case that `descriptorTypeCompatService` fails...